src/core/border_router/routing_manager.hpp

```diff
@@ -632,13 +633,16 @@ class RoutingManager : public InstanceLocator
             kDeprecating,
         };
 
+        void PublishAndAdvertise(void);
+        void Deprecate(void);
         void EnterAdvertisingState(void);
         void AppendCurPrefix(Ip6::Nd::RouterAdvertMessage &aRaMessage);
         void AppendOldPrefix(Ip6::Nd::RouterAdvertMessage &aRaMessage);
 
-        using ExpireTimer = TimerMilliIn<RoutingManager, &RoutingManager::HandleLocalOnLinkPrefixTimer>;
+        using ExpireTimer = TimerMilliIn<RoutingManager, &RoutingManager::HandleOnLinkPrefixManagerTimer>;
 
-        Ip6::Prefix mPrefix;
+        Ip6::Prefix mLocalPrefix;
+        Ip6::Prefix mDiscoveredPrefix;
         State       mState;
         TimeMilli   mExpireTime;
         Ip6::Prefix mOldPrefix;
```

Comment:
Nit: we probably want to make it clear that it is the old local on-link prefix, given that there is a new discovered on-link prefix now.
```suggestion
        Ip6::Prefix mOldLocalPrefix;
```

Reply:
Changed in new push.